server.py

```diff
@@ -1,10 +1,189 @@
-from flask import Flask, request, jsonify
+import datetime
+import os
+from pyexpat import model
+from time import timezone
+from flask import Flask, jsonify, request, render_template
+from flask_bcrypt import Bcrypt
+import jwt
+import secrets
+from flask_cors import CORS, cross_origin
+from models.User import db, User
+from sendgrid_config import send_reset_password_mail
 
 app = Flask(__name__)
+bcrypt = Bcrypt(app)
+cors = CORS(app)
+
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('DATABASE_URL').replace("postgres://", "postgresql://", 1) # sqlalchemy documents need the database name to include "ql" and this is the best way to remove it
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db.init_app(app)
+with app.app_context():
+    db.create_all()
+
+app.config['ACCESS_TOKEN_SECRET'] =  os.environ.get('ACCESS_TOKEN_SECRET')
+app.config['REFRESH_TOKEN_SECRET'] =  os.environ.get('REFRESH_TOKEN_SECRET')
+app.config['SENDGRID_EMAIL'] = os.environ.get('SENDGRID_EMAIL')
+app.config['SENDGRID_API_KEY'] = os.environ.get('SENDGRID_API_KEY')
 
 @app.route('/', methods=['POST', 'GET'])
 def hello_world():
     return jsonify({'success': 'Hello, World!'}), 200    
 
+def get_user_by_email(email):
+    user = db.session.query(User).filter(User.email == email)
+    if user.count() <= 0:
+        return None
+    if user.count() == 1:
+        for i in user:
+            return i.__dict__ 
+    if user.count() > 1:
+        return {'error':'voilates the unique ability!'}, 500
+
+def token_valid_check(token):
+    user = db.session.query(User).filter(User.reset_password_hash == token)
+    if user.count() <= 0:
+        return None
+    if user.count() == 1:
+        for i in user:
+            return i.__dict__
+    if user.count() > 1:
+        return {'error':'voilates the unique ability!'}, 500
+    
+@app.route('/signup', methods=['POST'])
```

Comment:
comments for all API paths below: 
1. move APIs to resources.py
2. move business logic (ie, creating user, updating user, getting user etc) to statemachine.py
3. reuse existing code as functions - don't rewrite